cmake/onnxruntime_providers.cmake

```diff
@@ -96,6 +96,73 @@ if (onnxruntime_USE_MKLDNN)
   set_target_properties(onnxruntime_providers_mkldnn PROPERTIES LINKER_LANGUAGE CXX)
 endif()
 
+if (onnxruntime_USE_TRT)
+  add_definitions(-DUSE_TRT=1)
+  add_definitions("-DONNX_ML=1")
+  add_definitions("-DONNX_NAMESPACE=onnx")
+  include_directories(${PROJECT_SOURCE_DIR}/external/protobuf)
+  set(CUDA_INCLUDE_DIRS ${CUDA_HOME}/include)
+  include_directories(${ONNXRUNTIME_ROOT}/../cmake/external/onnx)
+  if (WIN32)
+    set(OLD_CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS})
+    set(OLD_CMAKE_CUDA_FLAGS ${CMAKE_CUDA_FLAGS})
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /wd4996 /wd4244 /wd4267 /wd4099 /wd4551 /wd4505 /wd4515 /wd4706")
```

Comment:
we should document what warnings we are disabling.
also this works for Release flavor but not Debug. I had to also disable 4701 and 4805

Reply:
Also we should only disable the warnings for specific targets if possible. This is going to turn them off for everything we build isn't it?

Reply:
The warnings are not going to turn off everything. They only applied to onnx-tensorrt. After that, original setting is restored for other targets.
MM team updated the cmake file. Those two warnings 4701 and 4805 are disabled in Debug mode.